frappe/public/js/frappe/utils/diffview.js

```diff
@@ -0,0 +1,115 @@
+frappe.provide("frappe.ui");
+
+frappe.ui.DiffView = class DiffView {
+	constructor(doctype, fieldname, docname) {
+		this.dialog = null;
+		this.handler = null;
+		this.doctype = doctype;
+		this.fieldname = fieldname;
+		this.docname = docname;
+
+		this.dialog = this.make_dialog();
+		this.set_empty_state();
+		this.dialog.show();
+	}
+
+	make_dialog() {
+		const get_query = () => ({
+			query: "frappe.utils.diff.version_query",
+			filters: { docname: this.docname, ref_doctype: this.doctype },
+		});
+		const onchange = () => this.compute_diff();
+		let dialog = new frappe.ui.Dialog({
+			title: __("Compare Versions"),
+			fields: [
+				{
+					label: __("From version"),
+					fieldtype: "Link",
+					fieldname: "from_version",
+					options: "Version",
+					reqd: 1,
+					get_query,
+					onchange,
+				},
+				{
+					fieldtype: "Column Break",
+					fieldname: "cb",
+				},
+				{
+					label: __("To version"),
+					fieldtype: "Link",
+					fieldname: "to_version",
+					options: "Version",
+					reqd: 1,
+					get_query,
+					onchange,
+				},
+				{
+					fieldtype: "Section Break",
+					fieldname: "sb",
+				},
+				{
+					label: __("Diff"),
+					fieldtype: "HTML",
+					fieldname: "diff",
+				},
+			],
+			size: "large",
+		});
+		return dialog;
+	}
+
+	compute_diff() {
+		const from_version = this.dialog.get_value("from_version");
+		const to_version = this.dialog.get_value("to_version");
+		const fieldname = this.fieldname;
+
+		if (from_version && to_version) {
+			frappe
+				.xcall("frappe.utils.diff.get_version_diff", {
+					from_version,
+					to_version,
+					fieldname,
+				})
+				.then((data) => {
+					this.dialog.set_value("diff", this.prettify_diff(data));
+				});
+		} else {
+			this.set_empty_state();
+		}
+	}
+
+	prettify_diff(diff) {
+		let html = `
+		<style type="text/css">
+			.diffview {
+				font-family: monospace;
+				white-space: pre;
+				word-wrap: break-word;
+				color: #000000;
+			}
+		.diffview .insert {
+			background-color: #ddffdd;
+		}
+		.diffview .delete {
+			background-color: #ffdddd;
+		}
+		</style>
+		`;
```

Comment:
you can move this to the global SCSS file... also use CSS variables for colors and validate if dark mode looks good.